### ISSUE TITLE
`cargo -C` should have a long name

### ISSUE THREAD

**Issue by epage** [rust-lang/cargo#11696]
### Problem

#10952 added `-C` to solve #10098 and reduce #2930's impact but we couldn't get buy-in on a name and shipped it as-is to not block the feature on the long name as `-C` by itself has enough use within the community

### Proposed Solution

[From ehuss](https://github.com/rust-lang/cargo/pull/10952#issuecomment-1419778442):

> I don't have a strong opinion about the long name. `--cwd` or `--cd` seems more explicit as to what it does, and would lean slightly towards that, but other options seem fine. `-C` seems to be the most common for a short flag.
> 
> A quick survey:
> 
>   * make: `-C` or `--directory`
>   * git: `-C`
>   * Meson: `-C`
>   * ninja: `-C`
>   * hg: `--cwd`
>   * Gradle: `-p`, `--project-dir`?
>   * Julia: `--project`?
>   * Swift: `-working-directory`?
>   * dub: `--root`
>   * msbuild: passed as first argument
>   * npm: n/a
>   * Ivy: n/a?
>   * Maven: n/a?
>   * Bazel: n/a?
>   * Buck: n/a?
>   * go: [`-C`](https://pkg.go.dev/cmd/go#hdr-Compile_packages_and_dependencies)
>   * nuget: n/a?
>   * sbt: n/a?
>   * dart: [`-C` or `--directory`](https://dart.dev/tools/pub/cmd#--directorydir-or--c-dir)
> 
> Other command line tools with similar flags:
> 
>   * tar: `-C` or `--directory` (BSD tar has `--cd`)
>   * ls: `-d` or `--directory` to list directories
>     * rm: `-d` or `--dir` to remove directories

Plus the stdlib calls this "current dir" (`std::env::current_dir`, `std::process::Command::current_dir`) so we could name it `--current-dir`

**Comment by epage** [comment]
Non-blocker for this:  we could make it so all commonly used long names that are too CWD-specific or too generic to be used for anything else (e.g. `--directory`) are hidden aliases for `-C`, maybe with a message directing users to `-C`

**Comment by supreetsingh10** [comment]
Good evening, this issue is open, I wonder if anyone is taking this up? I think I can solve this issue. If this is available I can work on this?

**Comment by weihanglo** [comment]
Hi, @supreetsingh10. Thanks for being interested in helping this project.

The fix for this should be relatively straightforward. However, it currently needs some inputs from the Cargo team to proceed. As it is not really in a high priority, and it is more a bikeshedding problem, we haven't put too many efforts on deciding the name.

In the meantime, if you are willing to contribute to Cargo in general, I'd suggest looking into issue with label <https://github.com/rust-lang/cargo/labels/S-accepted>. We are happy to guide through and review those. Thank you!

See also [how our issue tracker works](https://doc.crates.io/contrib/issues.html).

**Comment by supreetsingh10** [comment]
Thank you very much, I would do that. I am also on the Zulip group and I will be in touch. Thank you :)
